app/controllers/application_controller.rb

```diff
@@ -151,4 +152,8 @@ def set_locale
   def sp_session
     session.fetch(:sp, {})
   end
+
+  def render_not_found
+    render template: 'pages/page_not_found', layout: false, status: 404, formats: :html
```

Comment:
Part of me wonders if we should render a 400 here instead? Maybe not b/c we don't have a template and all that setup for a 400?

Reply:
I have no strong opinions, we could even render the "not found" page with a 400 status code if you think that would be better?

Reply:
Maybe? Now that I'm looking at the bug this fixes, I'm thinking a 404 makes a bit more sense.

Reply:
ok we'll stick with 404 for now. Thanks!